### PR TITLE
Invalid referrencing runtime call casted to appropriate type.

### PR DESCRIPTION
**Pull request by CristinaCristescu**

### CODE REVIEW

File: interpreter/cling/include/cling/Interpreter/RuntimeUniverse.h [hunk 1]
```diff
@@ -176,7 +176,7 @@ extern "C" {
   /// run cling in -no-rtti mode.
   ///
 
-  void cling_runtime_internal_throwIfInvalidPointer(void* Sema,
+  void* cling_runtime_internal_throwIfInvalidPointer(void* Sema,
```
Comment (karies) [inline interpreter/cling/include/cling/Interpreter/RuntimeUniverse.h]: The decl returns `void*`, the impl `const void*`. Both should probably return `void*`.

File: interpreter/cling/lib/Interpreter/NullDerefProtectionTransformer.cpp [hunk 2]
```diff
@@ -238,7 +238,13 @@ namespace cling {
 
       Expr* call = m_Sema.ActOnCallExpr(S, unresolvedLookup, noLoc,
                                         args, noLoc).get();
-      return call;
+
+      SourceLocation loc;
+      TypeSourceInfo* TSI
+              = m_Context.getTrivialTypeSourceInfo(Arg->getType(), noLoc);
```
Comment (karies) [inline interpreter/cling/lib/Interpreter/NullDerefProtectionTransformer.cpp]: Can we maybe use the loc of the original pointer expression? Maybe also for the call expression itself?

**Comment by karies** [comment]
Does this pass roottest?

File: interpreter/cling/include/cling/Interpreter/RuntimeUniverse.h (hunk 1 above)
Comment (CristinaCristescu) [inline interpreter/cling/include/cling/Interpreter/RuntimeUniverse.h]: Done! Waiting for ctest now!

File: interpreter/cling/lib/Interpreter/Exception.cpp [hunk 3]
```diff
@@ -31,7 +31,7 @@ void cling_runtime_internal_throwIfInvalidPointer(void* Sema, void* Expr,
   else if (!cling::utils::isAddressValid(Arg))
     throw cling::InvalidDerefException(S, E,
           cling::InvalidDerefException::DerefType::INVALID_MEM);
-  return;
+  return const_cast<void*>(Arg);
```
Comment (karies) [inline interpreter/cling/lib/Interpreter/Exception.cpp]: Can you add a comment why we take `const void*` and return `void*` (namely to reduce the complexity in the calling AST nodes, at the expense of possibly doing a `T*` -> `const void*` -> `const_cast<void*>` -> `T*` round trip)?

Comment (CristinaCristescu) [inline interpreter/cling/lib/Interpreter/Exception.cpp]: Done!

**Comment by karies** [comment]
Merged, thanks!

**Pull request closed without merging**